corebehrt/functional/causal/checks.py

```
@@ -0,0 +1,47 @@
+    if data_end < data_start:
+        raise ValueError("data_end must be greater than data_start")
+
+    if "min_follow_up" not in time_windows:
+        raise ValueError("min_follow_up must be provided")
+    try:
+        pd.Timedelta(**time_windows["min_follow_up"])
+    except KeyError as e:
+        print(e)
+        raise ValueError(
+            "min_follow_up can be given in weeks, days, seconds, minutes, hours"
+        )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)